dashdataframe/__init__.py

```diff
@@ -242,6 +243,255 @@ def update_link(selectedData, ids):
         else:
             link = ""
         return link
+    
+
+    if add_umap:
+        @app.callback(
+        dash.dependencies.Output('umap-graphic', 'figure'),
+        [dash.dependencies.Input('umap_metrics', 'values'),
+        dash.dependencies.Input('umap_norm', 'value'),
+        dash.dependencies.Input('umap_button', 'n_clicks'),
+        dash.dependencies.Input('color_feature','value')
+        ])
+        
+        def update_umap_graph(selected_metrics, norm, umap_clicks, color_feature):
```

Comment:
I think this should probably get segregated into a umap specific file that gets imported rather than having it in this huge function.